<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1009</width>
    <height>907</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="BasicOpenGLView" name="oglwidget" native="true">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>700</width>
        <height>700</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="vertLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <property name="leftMargin">
       <number>10</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QGroupBox" name="modelsBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>250</width>
          <height>450</height>
         </size>
        </property>
        <property name="title">
         <string>Models</string>
        </property>
        <widget class="QListWidget" name="modelList">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>30</y>
           <width>231</width>
           <height>351</height>
          </rect>
         </property>
         <property name="focusPolicy">
          <enum>Qt::StrongFocus</enum>
         </property>
         <property name="selectionRectVisible">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QPushButton" name="addModel">
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>390</y>
           <width>75</width>
           <height>23</height>
          </rect>
         </property>
         <property name="text">
          <string>Add Model</string>
         </property>
        </widget>
        <widget class="QPushButton" name="removeModel">
         <property name="geometry">
          <rect>
           <x>150</x>
           <y>390</y>
           <width>78</width>
           <height>23</height>
          </rect>
         </property>
         <property name="text">
          <string>Remove Model</string>
         </property>
        </widget>
        <widget class="QCheckBox" name="orthoProj">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>420</y>
           <width>141</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Orthographic Projection</string>
         </property>
        </widget>
        <widget class="QCheckBox" name="postMult">
         <property name="geometry">
          <rect>
           <x>150</x>
           <y>420</y>
           <width>91</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Post Multiply</string>
         </property>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>250</width>
          <height>135</height>
         </size>
        </property>
        <property name="title">
         <string>RayTracer</string>
        </property>
        <widget class="QPushButton" name="traceButton">
         <property name="geometry">
          <rect>
           <x>14</x>
           <y>20</y>
           <width>221</width>
           <height>101</height>
          </rect>
         </property>
         <property name="text">
          <string>Trace Scene</string>
         </property>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="translationGB">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>250</width>
          <height>135</height>
         </size>
        </property>
        <property name="title">
         <string>Translation</string>
        </property>
        <widget class="QSlider" name="ZTransSlider">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>110</y>
           <width>211</width>
           <height>22</height>
          </rect>
         </property>
         <property name="minimum">
          <number>-100</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QSlider" name="XTransSlider">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>30</y>
           <width>211</width>
           <height>22</height>
          </rect>
         </property>
         <property name="minimum">
          <number>-100</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="singleStep">
          <number>0</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QLabel" name="xLabel_3">
         <property name="geometry">
          <rect>
           <x>110</x>
           <y>10</y>
           <width>61</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>X</string>
         </property>
        </widget>
        <widget class="QLabel" name="yLabel_3">
         <property name="geometry">
          <rect>
           <x>110</x>
           <y>50</y>
           <width>51</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Y</string>
         </property>
        </widget>
        <widget class="QSlider" name="YTransSlider">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>70</y>
           <width>211</width>
           <height>22</height>
          </rect>
         </property>
         <property name="minimum">
          <number>-100</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QLabel" name="zLabel_3">
         <property name="geometry">
          <rect>
           <x>110</x>
           <y>90</y>
           <width>61</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Z</string>
         </property>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="rotationGB">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>250</width>
          <height>140</height>
         </size>
        </property>
        <property name="title">
         <string>Rotation</string>
        </property>
        <widget class="QSlider" name="ZRotSlider">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>110</y>
           <width>211</width>
           <height>22</height>
          </rect>
         </property>
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>628</number>
         </property>
         <property name="singleStep">
          <number>4</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QLabel" name="zLabel">
         <property name="geometry">
          <rect>
           <x>110</x>
           <y>90</y>
           <width>51</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Z</string>
         </property>
        </widget>
        <widget class="QLabel" name="xLabel">
         <property name="geometry">
          <rect>
           <x>110</x>
           <y>10</y>
           <width>61</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>X</string>
         </property>
        </widget>
        <widget class="QLabel" name="yLabel">
         <property name="geometry">
          <rect>
           <x>110</x>
           <y>50</y>
           <width>51</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Y</string>
         </property>
        </widget>
        <widget class="QSlider" name="XRotSlider">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>30</y>
           <width>211</width>
           <height>22</height>
          </rect>
         </property>
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>628</number>
         </property>
         <property name="singleStep">
          <number>4</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QSlider" name="YRotSlider">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>70</y>
           <width>211</width>
           <height>22</height>
          </rect>
         </property>
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>628</number>
         </property>
         <property name="singleStep">
          <number>4</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <action name="actionAdd_Model">
   <property name="text">
    <string>Add Model...</string>
   </property>
  </action>
  <action name="actionRemove_Selected">
   <property name="text">
    <string>Remove Selected</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>BasicOpenGLView</class>
   <extends>QWidget</extends>
   <header>basicopenglview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
